sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionSuite.scala

```diff
@@ -1244,48 +1244,6 @@ class TypeCoercionSuite extends AnalysisTest {
     )
   }
 
-  test("BooleanEquality simplification") {
-    val be = TypeCoercion.BooleanEquality
-
-    ruleTest(be,
-      EqualTo(Literal(true), Literal(1)),
-      Literal(true)
-    )
```

Comment:
Do we have same tests for these?

I don't see the same for this one, for example, in `UnwrapCastInBinaryComparisonSuite`.

Reply:
Thanks, you are right that there is no case covered for comparing `Literal(true)` and `Literal(1)` combination. Now I feel I should separate out the work of removing the extra logic of `TypeCoercion` as a different PR with more tests. I will put them back for this PR